pkg/consensus/consensus.go

```diff
@@ -204,6 +204,21 @@ func (s *service) eventLoop() {
 			s.dbft.OnReceive(&msg)
 		case tx := <-s.transactions:
 			s.dbft.OnTransaction(tx)
+		case <-s.blockEvents:
+			// Drain events if there is more than one event queued.
+		drain:
+			for {
+				select {
+				case <-s.blockEvents:
```

Comment:
What about removing this loop and using `chan struct{}` of capacity 1 with non-blocking select on the send side?
It seems to have less overhead.